controllers/dockermachine_controller.go

```diff
@@ -198,16 +181,20 @@ func (r *DockerMachineReconciler) create(
 			return ctrl.Result{}, nil
 		}
 
-		// create a control plane
-		r.Log.Info("Creating a brand new cluster")
-		elb, err := getExternalLoadBalancerNode(c.Name, r.Log)
+		r.Log.Info("Creating a brand new controlplane node")
+		elb, err := actions.GetExternalLoadBalancerNode(c.Name)
 		if err != nil {
 			r.Log.Error(err, "Error getting external load balancer node")
 			return ctrl.Result{}, err
 		}
+		if elb == nil {
+			r.Log.Info("Cluster has no ELB yet, waiting for cluster network to be reconciled")
+			return ctrl.Result{RequeueAfter: time.Second * 30}, nil
+		}
+
 		lbipv4, _, err := elb.IP()
 		if err != nil {
-			r.Log.Error(err, "Error getting node IP address")
+			r.Log.Error(err, "Error getting IP address for ELB")
 			return ctrl.Result{}, err
 		}
 		controlPlaneNode, err := actions.CreateControlPlane(c.Name, machine.GetName(), lbipv4, *machine.Spec.Version, nil)
```

Comment:
Instead of using lbipv4 for the LB address here, we should probably set `Cluster.Status.APIEndpoints` in the DockerCluster controller, and consume that value here.

Reply:
Fixed.